dss/index/bundle.py

```diff
@@ -46,38 +49,59 @@ def _read_bundle_manifest(cls, replica: Replica, fqid: BundleFQID) -> dict:
         return manifest
 
     @classmethod
-    def _read_file_infos(cls, replica: Replica, fqid: BundleFQID, manifest: JSON) -> Mapping[str, JSON]:
+    def _read_file_infos(cls, replica: Replica, fqid: BundleFQID, manifest: JSON,
+                         context: LambdaContext) -> Mapping[str, JSON]:
         handle = Config.get_blobstore_handle(replica)
         bucket_name = replica.bucket
-        files_info = manifest[BundleMetadata.FILES]
-        assert isinstance(files_info, list)
+        files_info_original = manifest[BundleMetadata.FILES]
+        assert isinstance(files_info_original, list)
+        files_info = [file for file in files_info_original if file[BundleFileMetadata.INDEXED] is True]
+        files_remaining = len(files_info)
+        time_wait = 10  # time in sec
+
+        def enough_time(wait=0):
+            """
+            The max time_needed can be is 4 minutes, otherwise time_needed is based on the number of files left to
+            process
+            :return: bool
+            """
+            time_needed = min(240, files_remaining * time_wait)
```

Comment:
Why is this deriving the time needed to process a file from the sleep time after a failed attempt to load it. I think they should be independent. The former is `files_remaining * time_needed_to_load_a_file + elastic_search_overhead`. The latter should follow S3 recommendations: https://github.com/boto/botocore/blob/develop/botocore/data/s3/2006-03-01/waiters-2.json#L44